lib/wallet/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -1491,6 +1492,24 @@ toCardanoSimpleScript = \case
         Cardano.RequireTimeBefore Cardano.TimeLocksInSimpleScriptV2
         (O.SlotNo $ fromIntegral slot)
 
+toCardanoSimpleScriptV1
+    :: Script KeyHash
+    -> Cardano.SimpleScript Cardano.SimpleScriptV1
+toCardanoSimpleScriptV1 = \case
+    RequireSignatureOf (KeyHash _ keyhash) ->
+        case Cardano.deserialiseFromRawBytes
+            (Cardano.AsHash Cardano.AsPaymentKey) keyhash of
+                Just payKeyHash -> Cardano.RequireSignature payKeyHash
+                Nothing -> error "Hash key not valid"
+    RequireAllOf contents ->
+        Cardano.RequireAllOf $ map toCardanoSimpleScriptV1 contents
+    RequireAnyOf contents ->
+        Cardano.RequireAnyOf $ map toCardanoSimpleScriptV1 contents
+    RequireSomeOf num contents ->
+        Cardano.RequireMOf (fromIntegral num) $
+            map toCardanoSimpleScriptV1 contents
+    _ -> error "timelocks not available in SimpleScriptV1"
```

Comment:
👌 It seems that `toCardanoSimpleScriptV1` is only used from `updateSealedTx` in the `Shelley` era, which is unreachable in practice. This code should go away when we update all constraints from `IsShelleyBasedEra` to `IsRecentEra`.

Reply:
indeed